cluster-autoscaler/dynamic/config_fetcher.go

```diff
@@ -0,0 +1,44 @@
+package dynamic
+
+import (
+	"fmt"
+	kube_client "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_5"
+)
+
+// ConfigFetcher fetches the up-to-date dynamic configuration from the apiserver
+type ConfigFetcher struct {
+	configMapName string
+	namespace     string
+	kubeClient    kube_client.Interface
+	lastConfig    Config
+}
+
+// Builds a config fetcher from the parameters and dependencies
+func NewConfigFetcher(configMapName string, namespace string, kubeClient kube_client.Interface) *ConfigFetcher {
+	return &ConfigFetcher{
+		configMapName: configMapName,
+		namespace:     namespace,
+		kubeClient:    kubeClient,
+		lastConfig:    NewDefaultConfig(),
+	}
+}
+
+// Returns the config if it has changed since the last sync. Returns nil if it has not changed.
+func (c *ConfigFetcher) FetchConfigIfUpdated() (*Config, error) {
+	cm, err := c.kubeClient.Core().ConfigMaps(c.namespace).Get(c.configMapName)
```

Comment:
That's listed in my non-TODOs https://github.com/kubernetes/contrib/pull/2226#issue-197912602 for now.

If you'd like to do so, I'm ok with creating one with empty `nodeGroups` though.